logstash-core/src/main/java/org/logstash/ackedqueue/Queue.java

```diff
@@ -41,9 +41,12 @@
 
     protected volatile long unreadCount;
 
+    protected volatile long currentSize;
+
     private final CheckpointIO checkpointIO;
     private final PageIOFactory pageIOFactory;
     private final int pageCapacity;
```

Comment:
so if size is in bytes, it should probably be a `long` since the Java max integer is (2^31) - 1.